packages/sdk/src/tasks/finalize-withdrawal.ts

```diff
@@ -89,49 +94,67 @@ task('finalize-withdrawal', 'Finalize a withdrawal')
-      const currentStatus = await messenger.getMessageStatus(receipt)
-      console.log(
-        `Message status: ${
-          MessageStatus[currentStatus]
-        } (${await messenger.getLatestBlockNumber()} / ${receipt.blockNumber})`
-      )
-    }, 3000)
-
-    try {
-      await messenger.waitForMessageStatus(
-        receipt,
-        MessageStatus.READY_TO_PROVE
-      )
-    } finally {
-      clearInterval(proveInterval)
+    let currentStatus = await messenger.getMessageStatus(receipt)
+    if (currentStatus === MessageStatus.RELAYED) {
+      console.log('Withdrawal already proven and finalized')
+      return
     }
 
-    const proveTx = await messenger.proveMessage(txHash)
-    const proveReceipt = await proveTx.wait()
-    console.log('Prove receipt', proveReceipt)
-
-    const proveBlock = await hre.ethers.provider.getBlock(
-      proveReceipt.blockHash
-    )
-    const finalizationPeriodSeconds =
-      await messenger.getChallengePeriodSeconds()
-
-    console.log(`Withdrawal proven at ${proveBlock.timestamp}`)
-    console.log(`Finalization period is ${finalizationPeriodSeconds} seconds`)
+    if (currentStatus > MessageStatus.READY_TO_PROVE) {
+      console.log('Withdrawal already proven')
+    } else {
+      console.log('Waiting to be able to prove withdrawal')
+
+      const proveInterval = setInterval(async () => {
+        currentStatus = await messenger.getMessageStatus(receipt)
+        console.log(
+          `Message status: ${
+            MessageStatus[currentStatus]
+          } (${await messenger.getLatestBlockNumber()} / ${
+            receipt.blockNumber
+          })`
+        )
+      }, 3000)
+
+      try {
+        await messenger.waitForMessageStatus(
+          receipt,
+          MessageStatus.READY_TO_PROVE
+        )
+      } finally {
+        clearInterval(proveInterval)
+      }
 
-    const proveBlockFinalizedSeconds =
-      proveBlock.timestamp + finalizationPeriodSeconds
-    console.log(
-      `Waiting to be able to finalize withdrawal until ${proveBlockFinalizedSeconds}`
-    )
+      try {
+        const proveTx = await messenger.proveMessage(txHash)
+        const proveReceipt = await proveTx.wait()
+        console.log('Prove receipt', proveReceipt)
+
+        const proveBlock = await hre.ethers.provider.getBlock(
+          proveReceipt.blockHash
+        )
+        const finalizationPeriodSeconds =
+          await messenger.getChallengePeriodSeconds()
+
+        console.log(`Withdrawal proven at ${proveBlock.timestamp}`)
+        console.log(
+          `Finalization period is ${finalizationPeriodSeconds} seconds`
+        )
+
+        const proveBlockFinalizedSeconds =
+          proveBlock.timestamp + finalizationPeriodSeconds
+        console.log(
+          `Waiting to be able to finalize withdrawal until ${proveBlockFinalizedSeconds}`
+        )
+      } catch (e) {
+        console.log(`Error occurred during proving withdrawal: ${e}`)
```

Comment:
Could you add an error message that If you run this script on testnet, in most case, this happens when relayer has proved the transaction already.